application/src/test/java/org/thingsboard/server/transport/mqtt/mqttv3/credentials/BasicMqttCredentialsTest.java

```diff
@@ -114,11 +115,15 @@ public void testCorrectCredentials() throws Exception {
         testTelemetryIsDelivered(accessToken2Device, mqttTestClient5);
     }
 
-    @Test(expected = MqttSecurityException.class)
+    @Test(expected = MqttException.class)
     public void testCorrectClientIdAndUserNameButWrongPassword() throws Exception {
         // Not correct. Correct clientId and username, but wrong password
         MqttTestClient mqttTestClient = new MqttTestClient(CLIENT_ID);
-        mqttTestClient.connectAndWait(USER_NAME3, "WRONG PASSWORD");
+        try {
+            mqttTestClient.connectAndWait(USER_NAME3, "WRONG PASSWORD");
```

Comment:
Add Assert that checks we have not executed any code after connectAndWait and went straight to the catch block.